### PR TITLE
Added Trailblazer Avalon to QMK

### PR DESCRIPTION
**Pull request by mrnoisytiger**
## Description

Added Trailblazer Avalon to QMK

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/fjlabs/avalon/readme.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Avalon PCB by FJLaboratories
+
+The following is the QMK Firmware for the Avalon PCB, both Solder and Hotswap, for Trailblazer Keyboards.
+* Keyboard Maintainer: FJLaboratories
+* Hardware Supported: Avalon
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/fjlabs/avalon/readme.md:7]: Could you add instructions on how to enter the bootloader, to flash new firmware?  This is needed, as a new user may not be aware of how to enter to bootloader to flash new firmware.

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```
Could you add instructions on how to enter the bootloader, to flash new firmware?  This is needed, as a new user may not be aware of how to enter to bootloader to flash new firmware.

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/fjlabs/avalon/readme.md (hunk 1 above)
Comment (mrnoisytiger) [reply keyboards/fjlabs/avalon/readme.md:7]: Done.

**Review by drashna — APPROVED** [review]

**Comment by mrnoisytiger** [comment]
Bump. 😊

File: keyboards/fjlabs/avalon/info.json [hunk 2]
```diff
@@ -0,0 +1,88 @@
+{
+    "keyboard_name": "Avalon",
+    "url": "https://fjlaboratories.com/",
+    "maintainer": "FJLaboratories",
```
Comment (fauxpark) [inline keyboards/fjlabs/avalon/info.json:4]:
```suggestion
    "keyboard_name": "Avalon",
    "manufacturer": "Trailblazer Keyboards",
    "url": "https://fjlaboratories.com/",
    "maintainer": "FJLaboratories",
    "usb": {
        "vid": "0x7074",
        "pid": "0x0015",
        "device_version": "0.0.1"
    },
```

File: keyboards/fjlabs/avalon/config.h [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0x7074 // FJLaboratories
+#define PRODUCT_ID   0x0015 // Avalon
+#define DEVICE_VER   0x0001 // Version 1
+#define MANUFACTURER    Trailblazer Keyboards
+#define PRODUCT         Avalon
+
```
Comment (fauxpark) [inline keyboards/fjlabs/avalon/config.h:28]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**